core/common/src/main/java/alluxio/collections/NonUniqueFieldIndex.java

```diff
@@ -98,4 +111,70 @@ public T getFirst(Object value) {
     Set<T> all = mIndexMap.get(value);
     return all == null ? null : Iterables.getFirst(all, null);
   }
+
+  @Override
+  public Iterator<T> iterator() {
+    return new NonUniqueFieldIndexIterator();
+  }
+
+  @Override
+  public int size() {
+    int totalSize = 0;
+    for (ConcurrentHashSet<T> innerSet : mIndexMap.values()) {
```

Comment:
Is there a reason why you don't want to make this O(1), maybe with an `AtomicInteger`?